src/libraries/System.Linq/src/System/Linq/Sum.cs

```diff
@@ -60,11 +60,11 @@ public static partial class Enumerable
 
                 if (typeof(T) == typeof(long))
                 {
-                    return (TResult)(object)SumSignedIntegersVectorized(MemoryMarshal.Cast<T, long>(span));
+                    return (TResult)(object)SumSignedIntegersVectorized(Unsafe.BitCast<ReadOnlySpan<T>, ReadOnlySpan<long>>(span));
```

Comment:
Do these BitCasts generate same or better code compared to MemoryMarshal.Cast?

Reply:
Ever so slightly better, but nothing super meaningful (for this Sum method, the new version saves a mov). I mainly made the change in cases like this where it seemed to semantically be more appropriate: it's the same exact same type from/to, so BitCast seemed more appropriate than a more general function capable of casting between different types of different sizes. I don't feel strongly about it, though, if you'd prefer it be reverted.

Reply:
I was worried about code quality. I was not sure whether the JIT can handle the bitcast without spilling the value to stack. If the code quality is same or better, the change LGTM.

Reply:
Ok. I'll spot check some of the others, too.